packages/wrangler/src/__tests__/dev.test.tsx

```diff
@@ -231,6 +231,82 @@ describe("Dev component", () => {
       );
     });
   });
+
+  describe("custom builds", () => {
+    it("should run a custom build before starting `dev`", async () => {
+      writeWranglerToml({
+        build: {
+          command: `node -e "console.log('custom build'); require('fs').writeFileSync('index.js', 'export default { fetch(){ return new Response(123) } }')"`,
+        },
+      });
+
+      await runWrangler("dev index.js");
+
+      expect(readFileSync("index.js", "utf-8")).toMatchInlineSnapshot(
+        `"export default { fetch(){ return new Response(123) } }"`
+      );
+
+      // and the command would pass through
+      expect((Dev as jest.Mock).mock.calls[0][0].buildCommand).toEqual({
+        command:
+          "node -e \"console.log('custom build'); require('fs').writeFileSync('index.js', 'export default { fetch(){ return new Response(123) } }')\"",
+        cwd: undefined,
```

Comment:
oh hmm this doesn't match the type signature

Reply:
ok this is because of the way we do validation, it's fine I guess

Reply:
ok this is because of the way we do validation, it's fine I guess

Reply:
I'll keep it this way, we'll want to fail tests if and when the shape of this object changes